libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -437,6 +437,49 @@ protected async Task<DialogTurnResult> OnEndOfActionsAsync(SequenceContext seque
         protected async Task<RecognizerResult> OnRecognize(SequenceContext sequenceContext, CancellationToken cancellationToken = default(CancellationToken))
         {
             var context = sequenceContext.Context;
+            var noneIntent = new RecognizerResult()
+            {
+                Text = context.Activity.Text ?? string.Empty,
+                Intents = new Dictionary<string, IntentScore>()
+                    {
+                        { "None", new IntentScore() { Score = 0.0} }
+                    },
+                Entities = JObject.Parse("{}")
+            };
+            var text = context.Activity.Text;
+            if (context.Activity.Value != null)
+            {
+                var value = JObject.FromObject(context.Activity.Value);
+
+                // Check for submission of an adaptive card
+                if (string.IsNullOrEmpty(text) && value.Property("intent") != null)
+                {
+                    // Map submitted values to a recognizer result
+                    var recognized = new RecognizerResult() { Text = string.Empty };
+
+                    foreach (var property in value.Properties())
+                    {
+                        if (property.Name.ToLower() == "intent")
+                        {
+                            recognized.Intents[property.Value.ToString()] = new IntentScore() { Score = 1.0 };
+                        }
+                        else
+                        {
+                            if (recognized.Entities.Property(property.Name) == null)
+                            {
+                                recognized.Entities[property.Name] = new JArray(property.Value);
+                            }
+                            else
+                            {
+                                ((JArray)recognized.Entities[property.Name]).Add(property.Value);
```

Comment:
Are we 100% sure that thiswill be an array here? If not, this could throw InvalidCastException at runtime. If we are not sure, I'd swap to safe cast + validation or explicit exceptions for invalid cases.

Reply:
yes we are.

---
In reply to: [315918636](https://github.com/microsoft/botbuilder-dotnet/pull/2435#discussion_r315918636) [](ancestors = 315918636)